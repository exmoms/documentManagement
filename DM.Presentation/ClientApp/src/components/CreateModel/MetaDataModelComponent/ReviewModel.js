import React from "react";
import Typography from "@material-ui/core/Typography";
import ReactVirtualizedTable from "./TableView";
import { useTranslation } from "react-i18next";
import { makeStyles } from "@material-ui/core";

const useStyle = makeStyles(() => ({
  textAlign: {
    textAlign: "initial",
    padding: "20px",
  },
}));

export default function ReviewModel(props) {
  const { t } = useTranslation();
  const classes = useStyle();

  const attributeColumns = [
    { id: "metaDataAttributeName", label: t("nameofattribute"), minWidth: 170 },
    { id: "dataTypeID", label: t("type"), minWidth: 100 },
    {
      id: "isRequired",
      label: t("isrequired"),
      minWidth: 170,
      align: "center",
    },
  ];

  const selectedMetaDataModels_aggregateMetaDataModelsParts_Columns = [
    { id: "aggregateName", label: "Aggregation Name", minWidth: 170 },
    { id: "metaDataModelName", label: "Aggregated Model", minWidth: 100 },
  ];

  const compound_Columns = [
    { id: "caption", label: "caption", minWidth: 170 },
    { id: "isRequired", label: t("isrequired"), minWidth: 100 },
  ];

  var selectedMetaDataModels_aggregateMetaDataModelsParts = [
    {
      metaDataModelName: "",
      aggregateName: "",
    },
  ];

  if (
    props.value.selectedMetaDataModels.length ===
    props.value.aggregateMetaDataModelsParts.length
  ) {
    for (let i = 0; i < props.value.aggregateMetaDataModelsParts.length; i++) {
      selectedMetaDataModels_aggregateMetaDataModelsParts[i] = {
        metaDataModelName:
          props.value.selectedMetaDataModels[i].metaDataModelName,
        aggregateName:
          props.value.aggregateMetaDataModelsParts[i].aggregateName,
      };
    }
  }

  let compound;
  let aggregate;
  if (props.value.isCompound === true) {
    compound = (
      <>
        <p id="isCompound">
          <b>{t("compoundmodelincludes")}</b>
        </p>
        <ReactVirtualizedTable
          rows={props.value.compoundModels}
          columns={compound_Columns}
        />
      </>
    );
  } else {
    compound = (
      <p id="isCompound">
        <b>{t("notcompoundmodel")}</b>
      </p>
    );
  }

  if (props.value.isAggregated === true) {
    aggregate = (
      <>
        <p id="isAggregated">
          <b>{t("aggregatedmodelincludes")}</b>
        </p>
        <ReactVirtualizedTable
          rows={selectedMetaDataModels_aggregateMetaDataModelsParts}
          columns={selectedMetaDataModels_aggregateMetaDataModelsParts_Columns}
        />
      </>
    );
  } else {
    aggregate = (
      <p id="isAggregated">
        <b>{t("notaggregated")}</b>
      </p>
    );
  }

  const types = [
    t("booltype"),
    t("datetype"),
    t("decimal"),
    t("double"),
    t("int"),
    t("string"),
  ];

  var Attributes = [];

  for (let i = 0; i < props.value.metaDataAttribute.length; i++) {
    Attributes.push({
      dataTypeID: types[props.value.metaDataAttribute[i].dataTypeID - 1],
      metaDataAttributeName:
        props.value.metaDataAttribute[i].metaDataAttributeName,
      isRequired: props.value.metaDataAttribute[i].isRequired,
    });
  }

  return (
    <React.Fragment>
      <div className={classes.textAlign}>
        <Typography variant="h6" gutterBottom align="center">
          {t("modelreview")}
        </Typography>
        <hr />
        <p id="ModelName">
          <b>{t("modelname")}</b>: {props.value.metaDataModelName}
        </p>
        <p id="DocumentClassName">
          <b>{t("documentparagraph")}</b>:{" "}
          {props.value.documentClassSelected.documentClassName}
        </p>
        {compound}
        <br />
        {aggregate}
        <br />
        <b>{t("paragraphmetadata")}</b>:
        <br />
        <ReactVirtualizedTable rows={Attributes} columns={attributeColumns} />
      </div>
    </React.Fragment>
  );
}
