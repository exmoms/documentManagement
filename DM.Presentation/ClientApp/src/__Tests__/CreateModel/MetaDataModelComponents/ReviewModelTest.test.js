import React from "react";
import ReactDom from "react-dom";
import ReviewModel from "../../../components/CreateModel/MetaDataModelComponent/ReviewModel.js";
import { cleanup } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import TextField from "@material-ui/core/TextField";
import { act } from "react-dom/test-utils";
import Button from "@material-ui/core/Button";
afterEach(cleanup);
describe("Review Model", () => {
  test("renders without crashing", () => {
    const div = document.createElement("div");
    let MetaDataModelNameMocked = "";
    let DocumentClassSelectedMocked = [{}];
    let isCompoundMocked = false;
    let isAggregatedMocked = false;
    let AggregateMetaDataModelsPartsMocked = [
      { childMetaDataModelId: "", AggregateName: "" },
    ];
    let CompoundModelsMocked = [{ IsRequired: false, Caption: "" }];
    let SelectedMetaDataModelsMocked = [
      {
        metaDataModelName: "tested value 1",
        AggregateName: "aggregated name 1",
      },
    ];
    let MetaDataAttributeMock = [
      { Name: "", isRequired: false, dataTypeId: "" },
    ];
    ReactDom.render(
      <ReviewModel
        value={{
          metaDataModelName: MetaDataModelNameMocked,
          documentClassSelected: DocumentClassSelectedMocked,
          isCompound: isCompoundMocked,
          isAggregated: isAggregatedMocked,
          aggregateMetaDataModelsParts: AggregateMetaDataModelsPartsMocked,
          compoundModels: CompoundModelsMocked,
          selectedMetaDataModels: SelectedMetaDataModelsMocked,
          metaDataAttribute: MetaDataAttributeMock,
        }}
      />,
      div
    );
  });

  test("matches snapshot", () => {
    let MetaDataModelNameMocked = "Meta Data Model Name";
    let DocumentClassSelectedMocked = [{}];
    let isCompoundMocked = false;
    let isAggregatedMocked = false;
    let AggregateMetaDataModelsPartsMocked = [
      { childMetaDataModelId: "", AggregateName: "" },
    ];
    let CompoundModelsMocked = [{ IsRequired: false, Caption: "" }];
    let SelectedMetaDataModelsMocked = [
      {
        metaDataModelName: "tested value 1",
        AggregateName: "aggregated name 1",
      },
    ];
    let MetaDataAttributeMock = [
      { Name: "", isRequired: false, dataTypeId: "" },
    ];
    const wrapper = shallow(
      <ReviewModel
        value={{
          metaDataModelName: MetaDataModelNameMocked,
          documentClassSelected: DocumentClassSelectedMocked,
          isCompound: isCompoundMocked,
          isAggregated: isAggregatedMocked,
          aggregateMetaDataModelsParts: AggregateMetaDataModelsPartsMocked,
          compoundModels: CompoundModelsMocked,
          selectedMetaDataModels: SelectedMetaDataModelsMocked,
          metaDataAttribute: MetaDataAttributeMock,
        }}
      />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("test the functionality of the component", () => {
    let MetaDataModelNameMocked = "Meta Data Model Name";
    let DocumentClassSelectedMocked = {
      documentClassName: "Document Class Name 1",
    };
    let isCompoundMocked = false;
    let isAggregatedMocked = true;
    let AggregateMetaDataModelsPartsMocked = [
      { childMetaDataModelId: "", AggregateName: "" },
    ];
    let CompoundModelsMocked = [{ IsRequired: false, Caption: "" }];
    let SelectedMetaDataModelsMocked = [
      {
        metaDataModelName: "tested value 1",
        AggregateName: "aggregated name 1",
      },
    ];
    let MetaDataAttributeMock = [
      { Name: "", isRequired: false, dataTypeId: "" },
    ];
    const wrapper = shallow(
      <ReviewModel
        value={{
          metaDataModelName: MetaDataModelNameMocked,
          documentClassSelected: DocumentClassSelectedMocked,
          isCompound: isCompoundMocked,
          isAggregated: isAggregatedMocked,
          aggregateMetaDataModelsParts: AggregateMetaDataModelsPartsMocked,
          compoundModels: CompoundModelsMocked,
          selectedMetaDataModels: SelectedMetaDataModelsMocked,
          metaDataAttribute: MetaDataAttributeMock,
        }}
      />
    );
    expect(wrapper.find("#ModelName").text()).toBe(
      "modelname: Meta Data Model Name"
    );
    expect(wrapper.find("#DocumentClassName").text()).toBe(
      "documentparagraph: Document Class Name 1"
    );
    expect(wrapper.find("#isCompound").text()).toBe("notcompoundmodel");
    expect(wrapper.find("#isAggregated").text()).toBe(
      "aggregatedmodelincludes"
    );
  });
});
