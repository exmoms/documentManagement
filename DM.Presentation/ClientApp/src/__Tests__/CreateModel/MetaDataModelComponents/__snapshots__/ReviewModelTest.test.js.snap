// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Model matches snapshot 1`] = `
<Fragment>
  <div
    className="makeStyles-textAlign-1"
  >
    <WithStyles(ForwardRef(Typography))
      align="center"
      gutterBottom={true}
      variant="h6"
    >
      modelreview
    </WithStyles(ForwardRef(Typography))>
    <hr />
    <p
      id="ModelName"
    >
      <b>
        modelname
      </b>
      : 
      Meta Data Model Name
    </p>
    <p
      id="DocumentClassName"
    >
      <b>
        documentparagraph
      </b>
      :
       
    </p>
    <p
      id="isCompound"
    >
      <b>
        notcompoundmodel
      </b>
    </p>
    <br />
    <p
      id="isAggregated"
    >
      <b>
        notaggregated
      </b>
    </p>
    <br />
    <b>
      paragraphmetadata
    </b>
    :
    <br />
    <StickyHeadTable
      columns={
        Array [
          Object {
            "id": "metaDataAttributeName",
            "label": "nameofattribute",
            "minWidth": 170,
          },
          Object {
            "id": "dataTypeID",
            "label": "type",
            "minWidth": 100,
          },
          Object {
            "align": "center",
            "id": "isRequired",
            "label": "isrequired",
            "minWidth": 170,
          },
        ]
      }
      rows={
        Array [
          Object {
            "dataTypeID": undefined,
            "isRequired": false,
            "metaDataAttributeName": undefined,
          },
        ]
      }
    />
  </div>
</Fragment>
`;
